### ISSUE TITLE
"Attempt to free unreferenced scalar" with convert_blessed and overload

### ISSUE THREAD

**Issue by perlpunk** [rurban/Cpanel-JSON-XS#124]
@karenetheridge reported this issue over here:
https://github.com/mojolicious/json-validator/issues/149
I have nailed it down to this snippet, and it seems it was introduced in version 3.0202:
```
use 5.010;
use Cpanel::JSON::XS ();
say Cpanel::JSON::XS->VERSION;
my $coder = Cpanel::JSON::XS->new->allow_blessed->convert_blessed;
my $data = {nick => bless({}, 'BoolTestOk')};
my $json = $coder->encode($data);

package BoolTestOk;
use overload '""' => sub {1};
```
```
3.0202
Attempt to free unreferenced scalar: SV 0x7ff5fa82a8d8.
```
I'm on MacOS mojave (10.14.3) and perl 5.24.3

**Comment by rurban** [comment]
Nice. I saw it also but couldn't cut it down.

**Comment by perlpunk** [comment]
I will still have the mac for a couple of weeks and can test things, if that helps

**Comment by rurban** [comment]
Not needed, I can reliable repro it by myself.

**Comment by rurban** [comment]
Fixed with 4.10. Thanks for the nice reproducer

**Comment by karenetheridge** [comment]
Thanks - the JSON-Validator test looks great now!
